src/tsdb.c

```diff
@@ -129,7 +130,9 @@ GetSeriesResult GetSeries(RedisModuleCtx *ctx,
     *key = new_key;
 
     if (shouldDeleteRefs) {
-        deleteReferenceToDeletedSeries(ctx, *series);
+        const GetSeriesFlags deletion_flags =
+            flags & ~GetSeriesFlags_DeleteReferences | GetSeriesFlags_SilentOperation;
```

Comment:
Just verifying: what you want here is to turn the `GetSeriesFlags_DeleteReferences` off and `GetSeriesFlags_SilentOperation` on, right?

Reply:
yes.
`deleteReferenceToDeletedSeries` calls `GetSeries` with the flags it was provided, and we don't want to cause an infinite loop.
Some tests depend on `deleteReferenceToDeletedSeries` to be silent, so we must ensure it is on to prevent those tests from suddenly failing.

Reply:
Cool. Please add a comment explaining that.